### PR TITLE
/live split for channels going live

### LINKED ISSUE #1797: An in chat notification message for other channels going live
**What should be added?**
An in chat notification message across all channels that tells you if a channel in the "channels going live" list went live.

**Why should it be added?**
Twitch live notifications is usually much delayed than chatterinos live alerts, an informative live message in the chat for any channel in the list that goes live might be a good feature. I am imagining it would look the same as "hh:mm connected" or "hh:mm Now hosting shroud." messages and it would be something like "hh:mm shroud went live." to be displayed in forsens chat for example along with the sound alert.

### PR DESCRIPTION
**Pull request by apa420**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

Adds a /live split for when channels go live.
Done:
- [x] A /live split
- [x] Nice looking message
- [x] Highlights and such
Closes #1797

### CODE REVIEW

File: src/widgets/dialogs/SelectChannelDialog.cpp [hunk 1]
```diff
@@ -404,10 +426,12 @@ bool SelectChannelDialog::EventFilter::eventFilter(QObject *watched,
                 this->dialog->ui_.twitch.whispers->setFocus();
                 return true;
             }
-            else
+            else if (widget == this->dialog->ui_.twitch.live)
             {
-                widget->nextInFocusChain()->setFocus();
```
Comment (fourtf) [inline src/widgets/dialogs/SelectChannelDialog.cpp:409]: why was this removed?

Comment (apa420) [reply src/widgets/dialogs/SelectChannelDialog.cpp:409]: It's not removed...
It's on line 434 now.
I added logic for going from the bottom element to the top one.

File: src/messages/SharedMessageBuilder.cpp [hunk 2]
```diff
@@ -148,6 +148,40 @@ void SharedMessageBuilder::parseHighlights()
         // Don't check for any other highlight phrases.
         return;
     }
+    if (this->message().flags.has(MessageFlag::Live))
+        qDebug() << "I am here";
```
Comment (fourtf) [inline src/messages/SharedMessageBuilder.cpp:152]: FeelsDOEnkMan

File: src/providers/twitch/TwitchMessageBuilder.cpp [hunk 3]
```diff
@@ -1212,4 +1212,16 @@ void TwitchMessageBuilder::appendChannelPointRewardMessage(
...
 
+void TwitchMessageBuilder::liveMessage(const QString &channelName,
+                                       MessageBuilder *builder)
+{
+    builder->emplace<TimestampElement>();
+    builder->emplace<TextElement>(channelName + " is live!",
+                                  MessageElementFlag::Text, MessageColor::Text,
+                                  FontStyle::ChatMediumBold);
+    builder->message().flags.set(MessageFlag::Live);
+    //builder->message()->
```
Comment (fourtf) [inline src/providers/twitch/TwitchMessageBuilder.cpp:1223]: remove comment

File: src/providers/twitch/TwitchChannel.cpp [hunk 4]
```diff
@@ -522,9 +522,14 @@ void TwitchChannel::setLive(bool newLiveStatus)
...
                     }
                 }
-                auto live =
-                    makeSystemMessage(this->getDisplayName() + " is live");
-                this->addMessage(live);
+                MessageBuilder builder;
+                TwitchMessageBuilder::liveMessage(this->getDisplayName(),
+                                                  &builder);
+                this->addMessage(builder.release());
+                //auto live =
```
Comment (fourtf) [inline src/providers/twitch/TwitchChannel.cpp:529]: remove comment

**Comment by fourtf** [comment]
Looks good, however it doesn't seem to compile on linux/macos CI. If that's fixed then I'd be glad to merge!

File: src/controllers/notifications/NotificationController.cpp [hunk 5]
```diff
@@ -102,9 +102,9 @@ void NotificationController::playSound()
     static QUrl currentPlayerUrl;
 
     QUrl highlightSoundUrl =
-        getSettings()->notificationCustomSound
+        getSettings()->notificationSoundUrl.getValue() != ""
```
Comment (23rd) [inline src/controllers/notifications/NotificationController.cpp:105]:
```suggestion
        !getSettings()->notificationSoundUrl.getValue().isEmpty()
```

File: src/singletons/Settings.hpp [hunk 6]
```diff
@@ -261,13 +261,8 @@ class Settings : public ABSettings, public ConcurrentSettings
     /// Notifications
     BoolSetting notificationFlashTaskbar = {"/notifications/enableFlashTaskbar",
                                             false};
-    BoolSetting notificationPlaySound = {"/notifications/enablePlaySound",
```
Comment (fourtf) [inline src/singletons/Settings.hpp:264]: why was this changed

File: src/providers/twitch/IrcMessageHandler.cpp [hunk 7]
```diff
@@ -270,12 +270,6 @@ void IrcMessageHandler::addMessage(Communi::IrcMessage *_message,
                 server.mentionsChannel->addMessage(msg);
             }
         }
-
-        chan->addMessage(msg);
```
Comment (fourtf) [inline src/providers/twitch/IrcMessageHandler.cpp:274]: where does it add the message to the channel now?

File: CHANGELOG.md [hunk 8]
```diff
@@ -2,6 +2,8 @@
 
 ## Unversioned
 
+- Major: New split for channels going live! /live
+
```
Comment (pajlada) [inline CHANGELOG.md:6]: Please add a changelog entry for the UX fix with tabbing in the "Select channel" dialog

**Review by pajlada — CHANGES_REQUESTED** [review]

**Review by pajlada — APPROVED** [review]
